### PR TITLE
Implement `nsc expose-lb`

### PR DESCRIPTION
**Pull request by n-g**
Fixes NSL-977

### CODE REVIEW

**Comment by n-g** [comment]
Validated e2e:

```
% nsc expose-lb 9nb7lm0f126ge --service nginx-service
Exported port 80 from default/nginx-service:
  https://5qtj550-9nb7lm0f126ge.fra1.nscluster.cloud
```

**Comment by n-g** [comment]
Fixes NSL-977

**Comment by n-g** [comment]
Fixes NSL-977

File: internal/cli/cmd/cluster/exposelb.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	"namespacelabs.dev/foundation/internal/console"
+	"namespacelabs.dev/foundation/internal/fnerrors"
+	"namespacelabs.dev/foundation/internal/providers/nscloud/api"
+	"namespacelabs.dev/foundation/internal/providers/nscloud/ctl"
+	"namespacelabs.dev/foundation/std/tasks"
+)
+
+func NewExposeLoadBalancerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "expose-lb [cluster-id]",
```
Comment (hugosantos) [inline internal/cli/cmd/cluster/exposelb.go:27]: what about:
`expose kubernetes`
`expose container`

and `expose` would be an alias of `expose container`

File: internal/cli/cmd/cluster/exposelb.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		Aliases: []string{"expose-load-balancer"},
+		Short:   "Opens a public ingress to the specified service load balancer.",
+		Args:    cobra.MaximumNArgs(1),
+	}
+
+	name := cmd.Flags().String("name", "", "If specified, set the name of the exposed ingress. Only permitted when exposing a single port. By default, ingress names are generated.")
+	namespace := cmd.Flags().String("namespace", "default", "Namespace of the service load balancer to export.")
```
Comment (hugosantos) [inline internal/cli/cmd/cluster/exposelb.go:34]: I would not set a default.

File: internal/cli/cmd/cluster/exposelb.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	service := cmd.Flags().String("service", "", "Name of the service load balancer to export.")
+	port := cmd.Flags().Int32("port", 0, "Which port of service load balancer to export.")
```
Comment (hugosantos) [inline internal/cli/cmd/cluster/exposelb.go:36]: `Which exported Load Balancer port to expose.`

File: internal/cli/cmd/cluster/exposelb.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+			return nil, fnerrors.New("failed to create kubernetes client: %w", err)
+		}
+
+		svc, err := cli.CoreV1().Services(ns).Get(ctx, service, metav1.GetOptions{})
+		if err != nil {
+			return nil, fnerrors.InvocationError("kubernetes", "failed to query service %q: %w", service, err)
+		}
+
+		if svc.Spec.Type != corev1.ServiceTypeLoadBalancer {
+			return nil, fnerrors.New("service %q is not a load balancer (found type %q)", service, svc.Spec.Type)
```
Comment (hugosantos) [inline internal/cli/cmd/cluster/exposelb.go:97]: service %q is not of type %s

and I'd pass type=corev1.ServiceTypeLoadBalancer

**Review by hugosantos — APPROVED** [review]

File: internal/cli/cmd/cluster/exposelb.go [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+		Short:   "Opens a public ingress to the specified service load balancer.",
+		Args:    cobra.MaximumNArgs(1),
+	}
+
+	name := cmd.Flags().String("name", "", "If specified, set the name of the exposed ingress. Only permitted when exposing a single port. By default, ingress names are generated.")
+	namespace := cmd.Flags().String("namespace", "default", "Namespace of the service load balancer to export.")
+	service := cmd.Flags().String("service", "", "Name of the service load balancer to export.")
+	port := cmd.Flags().Int32("port", 0, "Which port of service load balancer to export.")
+
+	ip := cmd.Flags().String("ip", "", "Which ip of service load balancer to export with the ingress.")
```
[the remark this answers is not in the document]
Comment (n-g) [reply internal/cli/cmd/cluster/exposelb.go]: From what I can tell, we already validate the CIDR https://github.com/namespacelabs/internal/blob/75826581c8ac9ba4471276a71cd2519e0e07136f/vm/host/vmhost/ingress.go#L229-L230 
https://github.com/namespacelabs/internal/blob/75826581c8ac9ba4471276a71cd2519e0e07136f/vm/host/vmhost/config.go#L46-L47

**Pull request merged**